pkg/codegen/python/gen.go

```diff
@@ -1193,7 +1193,7 @@ func (mod *modContext) genResource(res *schema.Resource) (string, error) {
 
 		if hasStateInputs {
 			for _, prop := range res.StateInputs.Properties {
-				pname := PyName(prop.Name)
+				pname := InitParamName(prop.Name)
```

Comment:
I think we also need to use `InitParamName` when assigning the param to `__props__.__dict__` below.